transport/tchannel/outbound.go

```diff
@@ -176,9 +193,28 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		return nil, err
 	}
 
-	buf := bytes.NewBuffer(make([]byte, 0, _defaultBufferSize))
-	if _, err = buf.ReadFrom(resBody); err != nil {
-		return nil, err
+	var body io.ReadCloser
+	var bodySize int
+	if reuseBuffer {
+		buffer := bufferpool.NewAutoReleaseBuffer()
+		if _, err = buffer.ReadFrom(resBody); err != nil {
+			return nil, err
+		}
+		body = readerCloser{
+			Reader: bytes.NewReader(buffer.Bytes()),
+			Closer: buffer,
+		}
```

Comment:
```suggestion
		body = buffer
```

We can assign buffer directly on body as it implements close method

Reply:
Doing that would allow callers to modify the response body. Ref #2233